### PR TITLE
feat: add movie review pt2

### PR DESCRIPTION
**Pull request by jamesrp13**

### CODE REVIEW

File: content/movie-review-pt2.md [hunk 1]
```diff
@@ -10,8 +10,9 @@
 
 # TL;DR
 
-- Creating an account requires that we calculate the space and rent to allocate for a new account
-- A Program Derived Address (PDA) is derived from a program Id and an optional list of seeds
+- Program state is stored in external accounts rather than in the program itself
+- A Program Derived Address (PDA) is derived from a program Id and an optional list of seeds and subsequently used as the address for a storage account
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:14]: sugg: add a comma, or make a new sentence... too many "and"s, e.g. 

A Program Derived Address (PDA) is derived from a program Id and an optional list of seeds. Once derived, PDAs are subsequently used as the address for a storage account.

or,

A Program Derived Address (PDA) is derived from a program Id and an optional list of seeds, and is subsequently used as the address for a storage account

File: content/movie-review-pt2.md [hunk 2]
```diff
@@ -10,8 +10,9 @@
 
 # TL;DR
 
-- Creating an account requires that we calculate the space and rent to allocate for a new account
-- A Program Derived Address (PDA) is derived from a program Id and an optional list of seeds
+- Program state is stored in external accounts rather than in the program itself
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:13]: sugg: "another" account, rather than "external" account (which may be interpreted as "off-chain"?)  I am judging myself negatively for even bringing this one up...

File: content/movie-review-pt2.md [hunk 3]
```diff
@@ -25,13 +26,15 @@ In this lesson we will learn the basics of state management for a Solana program
...
 
-Under the hood, account data is represented as a byte array, just as it is with instruction data. Since we'll be using account data to manage our program's state, one of the first things we can do to make managing state easier is to represent it as a Rust data type. 
+When writing a program in Rust, we typically create this "format" by defining a Rust data type. If you went through the [Create a Basic Program: Part 1 lesson](basic-program-pt-1.md), this is very similar to what we did when we created an enum to represent discrete instructions.
 
-While this type should reflect the structure of your data, for most use cases a simple struct is sufficient. For example, a note-taking program that stores notes in their own accounts would likely have data for a title, body, and maybe an ID of some kind. We could create a struct to represent that as follows:
+While this type should reflect the structure of your data, for most use cases a simple struct is sufficient. For example, a note-taking program that stores notes in separate accounts would likely have data for a title, body, and maybe an ID of some kind. We could create a struct to represent that as follows:
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:37]: I dig the flow a lot better here for how we discuss state

File: content/movie-review-pt2.md [hunk 4]
```diff
@@ -43,13 +46,9 @@ struct NoteState {
 
 ### Using Borsh for Serialization and Deserialization
 
-To access the data store on an account, we must serialize and deserialize the data. 
-1. Serialization is the process of converting an object into a bytes array (an area of memory containing a continuous sequence of bytes). 
-2. Deserialization is the process of reconstructing an object from a byte array. 
-
-For our use case, the object is simply the data we want to store. Writing to the data field of an account requires that we serialization the data we want to store. To read the data from an existing account, we must deserialized the account data from bytes into readable objects. We can then update this (deserialized) data and serialize the data into the account again. 
+Just as with instruction data, we need a mechanism for converting from our Rust data type to a byte array, and vis versa. **Serialization** is the process of converting an object into a byte array. **Deserialization** is the process of reconstructing an object from a byte array.
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:49]: - typo "vis versa" --> vice versa
- 2¢ "object to/from byte array" feels dry to me... which is good and clear, which adequate to our audience, but Ivan 'splained it to me as "human readable" <> byte Buffer.  sugg: add something to the effect of, "another way to think of it is converting to and from something human readable to a byte array."

File: content/movie-review-pt2.md [hunk 5]
```diff
@@ -133,9 +98,9 @@ let rent_lamports = rent.minimum_balance(account_len);
...
+In addition to the seeds you provide for deriving a PDA, the `find_program_adress` function will provide one additional "bump seed." What makes PDAs unique from other Solana account addresses is that they do not hae a corresponding secret key. This ensures that only the program that owns the address can sign on behalf of the PDA. When the `find_program_addres` function attempts to derive a PDA using the provided seeds, it passes in the number 255 as the "bump seed." If the resulting address is invalid (i.e. has a corresponding secret key), then the function decreases the bump seed by 1 and derives a new PDA with that bump seed. Once a valid PDA is found, the function returns both the PDA and the bump that was used to derive the PDA.
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:101]: typo "hae" --> have (*2nd sentence)

File: content/movie-review-pt2.md [hunk 6]
```diff
@@ -133,9 +98,9 @@ let rent_lamports = rent.minimum_balance(account_len);
...
 
-For our note-taking program, we will use the note creator's publickey and the id as the optional seeds to derive the PDA. Deriving the PDA this way allows us to deterministicly find the account for each note.
+For our note-taking program, we will use the note creator's public key and the ID as the optional seeds to derive the PDA. Deriving the PDA this way allows us to deterministicly find the account for each note.
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:103]: typo "deterministicly" --> deterministically

File: content/movie-review-pt2.md [hunk 7]
```diff
@@ -200,26 +165,69 @@ msg!("borrowed account data");
...
 ```
 
 ### Serialize Account Data
 
-The `account_data` variable now holds the deserialized data in the format of our `NoteState` struct updated with the user’s inputs. We now need to save the data on the `account_data` variable to the data field on our actual `note_pda_account`. We do this by borrowing a mutable reference to the data of the `note_pda_account` and then serializing (convert to bytes) the content of the `account_data` variable to the data field of the `note_pda_account`. 
+Once the Rust instance representing the account's data has been updated with the appropriate values, you can "save" the data on the account.
 
-This effectively saves the updated `account_data` variable to the data field of the new account. Now when a user fetches the `note_pda_account` and deserializes (convert from bytes) the data, it will display the updated data we’ve serialized into the account.
+This is done with the `serialize` function on the instance of the Rust type you created. You'll need to pass in a mutable reference to the account data. The syntax here is tricky, so don't worry if you don't understand it completely. Borrowing and references are two of the trickier concepts in Rust.
```
Comment (mixelpixel) [inline content/movie-review-pt2.md:175]: love this

**Review by mixelpixel — COMMENTED** [review]
@jamesrp13 a review of your edits - all in all, great improvements, reads really well, and the flow feels a lot more directed <3

File: content/movie-review-pt2.md (hunk 4 above)
Comment (jamesrp13) [reply content/movie-review-pt2.md:49]: That's not technically correct though. It happens to be somewhat human readable but technically it's about converting back and forth between formats. Neither of the formats necessarily have to be human readable

File: content/movie-review-pt2.md (hunk 6 above)
Comment (jamesrp13) [reply content/movie-review-pt2.md:103]: Wow haha that's a bad one

**Review by ZYJLiu — APPROVED** [review]

**Comment by mixelpixel** [comment]
@jamesrp13 one more typo fix: https://github.com/Unboxed-Software/solana-course/pull/70

**Review by ZYJLiu — APPROVED** [review]
